server.js

```diff
@@ -1,35 +1,180 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import dotenv from "dotenv";
+import listEndpoints from "express-list-endpoints";
 
 // If you're using one of our datasets, uncomment the appropriate import below
 // to get started!
 // import avocadoSalesData from "./data/avocado-sales.json";
 // import booksData from "./data/books.json";
 // import goldenGlobesData from "./data/golden-globes.json";
 // import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+import topMusicData from "./data/top-music.json";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+dotenv.config()
+const mongoUrl = process.env.MONGO_URL || `mongodb+srv://${process.env.MONGO_USER}:${process.env.MONGO_PW}@cluster0.lmkbhok.mongodb.net/mongoAPI?retryWrites=true&w=majority`;
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
-
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
+
+const Song = mongoose.model("Song", {
+    id: Number,
+    trackName: String,
+    artistName: String,
+    genre: String,
+    bpm: Number,
+    energy: Number,
+    danceability: Number,
+    loudness: Number,
+    liveness: Number,
+    valence: Number,
+    length: Number,
+    acousticness: Number,
+    speechiness: Number,
+    popularity: Number
+})
+if(process.env.RESET_DB){
+  const resetDataBase = async () => {
+    await Song.deleteMany();
+    topMusicData.forEach(singleSong => {
+      const newSong = new Song (singleSong);
+      newSong.save();
+    })
+
+  }
+  resetDataBase();
+}
 const port = process.env.PORT || 8080;
 const app = express();
 
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({
+      error: `Service unavailable`
+    })
+  }
+})
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send({ "Routes": listEndpoints(app)});
 });
 
+
+/// All songs
+app.get("/songs", async (req, res) => {
+  const allSongs = await Song.find({});
+    res.status(200).json({
+    success: true,
+    body: allSongs
+    });
+  });
+
+  //sorts all songs based on popularity rating
+  app.get("/songs/popularity", async (req, res) => {
+    try {
+      let popular = await Song.find()
+      popular = popular.sort((a, b) => b.popularity - a.popularity)
+      res.json({ length: popular.length, data: popular })
+    } catch {
+      res.status(400).json ({ error: `invalid request` })
+    }
+  })
```

Comment:
I really like that you managed to sort the Song list. I tried to do it but couldn't get it to work so well done.